src/fileActions.jsx

```diff
@@ -603,3 +603,30 @@
                     return file;
             }, exc => console.error(`Adding file ${file} failed: ${exc.toString()}`));
 };
+
+export const downloadFile = (currentPath, selected) => {
+    const query = window.btoa(JSON.stringify({
+        payload: "fsread1",
+        binary: "raw",
+        path: `${currentPath}/${selected.name}`,
+        superuser: "try",
+        external: {
+            "content-disposition": `attachment; filename="${selected.name}"`,
+            "content-type": "application/octet-stream",
+            "content-length": Number(selected.size),
+        }
+    }));
+
+    const prefix = (new URL(cockpit.transport.uri("channel/" + cockpit.transport.csrf_token))).pathname;
+    const url = prefix + "?" + query;
+    return new Promise((resolve, reject) => {
+        const iframe = document.createElement("iframe");
+        iframe.setAttribute("src", url);
+        iframe.setAttribute("hidden", "hidden");
+        iframe.addEventListener("load", event => {
+            document.body.removeChild(iframe);
+            resolve();
+        });
+        document.body.appendChild(iframe);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-175-20240126-132259-b04878fd-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)